File: app/server/notify/channels/pushme.ts
/**
 * PushMe 推送渠道
 */
import Logger from '../../utils/logger';
import { NotifyChannel, NotifyResult } from '../types';
import { $ } from '../httpClient';
import { hasConfig, getConfig } from '../config';

const logger = Logger.child({ channel: 'pushme' });

export const pushmeChannel: NotifyChannel = {
    name: 'pushme',
    enabled: false,

    async send(text: string, desp: string): Promise<NotifyResult> {
        return new Promise((resolve) => {
            if (!hasConfig('PUSHME_KEY')) {
                resolve({ success: false, message: 'PUSHME_KEY 未配置' });
                return;
            }

            const PUSHME_KEY = getConfig('PUSHME_KEY');
            const url = `https://push.i-i.me?pushkey=${PUSHME_KEY}`;

            const body = {
                title: text,
                content: desp
            };

            $.post({ url, json: body }, (err, _resp, data) => {
                if (err) {
                    logger.error({ err }, 'PushMe 发送失败');
                    resolve({ success: false, error: err as Error });
                } else {
                    const result = data as { code?: number; message?: string };
                    if (result.code === 200 || result.code === 0) {
                        logger.info('PushMe 发送成功');
                        resolve({ success: true, message: '发送成功' });
                    } else {
                        logger.warn({ body: data }, 'PushMe 发送失败');
                        resolve({ success: false, message: result.message || '发送失败' });
                    }
                }
            });
        });
    }
};

if (hasConfig('PUSHME_KEY')) {
    pushmeChannel.enabled = true;
}

export default pushmeChannel;

File: app/server/internal/notify/channels/pushme.go
package channels

import (
	"log/slog"

	"github.com/sushazhi/fnos-logmanager/internal/notify"
)

type PushMe struct {
	enabled bool
}

func (c *PushMe) Name() string {
	return "pushme"
}

func (c *PushMe) Enabled() bool {
	return c.enabled
}

func (c *PushMe) SetEnabled(enabled bool) {
	c.enabled = enabled
}

func (c *PushMe) Send(text, desp string) notify.NotifyResult {
	if !notify.HasConfig("PUSHME_KEY") {
		return notify.NotifyResult{Success: false, Message: "PUSHME_KEY 未配置"}
	}

	pushmeKey := notify.GetConfig("PUSHME_KEY")
	url := "https://push.i-i.me?pushkey=" + pushmeKey

	body := map[string]string{
		"title":   text,
		"content": desp,
	}

	resp, err := notify.HTTPPost(url, notify.HttpRequestOptions{
		JSON: body,
	})
	if err != nil {
		slog.Error("PushMe 发送失败", "error", err)
		return notify.NotifyResult{Success: false, Message: err.Error(), Err: err}
	}

	if resp.StatusCode == 200 {
		slog.Info("PushMe 发送成功")
		return notify.NotifyResult{Success: true, Message: "发送成功"}
	}

	slog.Warn("PushMe 发送失败", "status", resp.StatusCode, "body", resp.Body)
	return notify.NotifyResult{Success: false, Message: "发送失败"}
}

func init() {
	ch := &PushMe{}
	if notify.HasConfig("PUSHME_KEY") {
		ch.SetEnabled(true)
	}
	notify.Registry.Register(ch)
}
